docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,281 @@
+
+      The value type of this field is STRING.
+
+    - **`compatibility score field`**
+
+      This OPTIONAL property describes the score of the compatibility set.
+
+      There is only one compatibility score filed allowed in one compatibility set.
+
+      The highest score of the compatibility set means this compatibility set is the most optimal one.
+
+      If this field is not present, the score of this compatibility set is 0.
+
+      If compatibility sets have the same score, that means any of these compatibility sets pass the compatibility check is successful in matching the compatibility requirements.
+
+      The compatibility-score is used for selecting optimal image or optimal node in descending order for those compatibility sets pass the compatibility check.
+
+      The value type of this field is INT.
+
+    - **`annotation field`**
+
+      This OPTIONAL property is a field to describe human readable information regarding to the compatibility set.
+
+      The value type of this field is STRING.
+
+## Artifact example of use
+
+### Example - image is compatible with different hardware combinations
+
+```json
+{
+  "compatibilities": [
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-15B3-020D.present": "true",
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-1002-67ff.present": "true",
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+### Example - image is compatible with optimal hardware
+
+```json
+{
+  "compatibilities": [
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AMXFP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-019D.present": "true",
+      "compatibility-score": 50,
+    },
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-020D.present": "true",
+      "compatibility-score": 100,
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
```

Comment:
I like the short am simple descriptiveness of it.

One question: Will there be operators to version numbers? So that we can express greater than and exact version matches? Or has minimal version limits had problems in the past?

Reply:
I didn't notice the different brackets, hence my assumption it was exact version matches and not version expressions. In most places I work with minimal version required rather than a fixed version. So I was curious if that leads to problems in the linux world. Would a single version express `>= $VERSION` in this case?

Reply:
This is nit-picking, but I think strings like `>=4.19,<5.16` would be easier for users to use and easier to parse (as well as allowing for more complicated rules, though there is an argument that you don't want to allow super-complicated rules).

Reply:
> This is nit-picking, but I think strings like `>=4.19,<5.16` would be easier for users to use and easier to parse (as well as allowing for more complicated rules, though there is an argument that you don't want to allow super-complicated rules).

I am open for the format. we have to consider discontinous range(it happened), for example, maybe "4.19 <= version < 5.16, version = 5.17, version >= 6.0" it's also fine. yes, no need to allow super complicated rules.

Reply:
> This is nit-picking, but I think strings like `>=4.19,<5.16` would be easier for users to use and easier to parse (as well as allowing for more complicated rules, though there is an argument that you don't want to allow super-complicated rules).

As mentioned in the latest commit, "The proposal itself does not intend to define concrete compatibility field labels, it should be a task for separate proposals". More elaboration of field value definition in future proposals